### ISSUE TITLE
question: example of released web apps which use Aurelia

### ISSUE THREAD

**Issue by opcodewriter** [aurelia/framework#333]
I heard Rob mentioned for quite a while now that there are companies which Aurelia in production for many months, but he cannot disclose their name, which can be understandable sometimes. 

But are there any examples of web apps which uses Aurelia in production and whose names can be given? 

I'm looking for:
1. a real product/service, so not a demo or POC. Also something which is like an internal product doesn't help, because I want to check it out.
2. some complexity in the data flow, must have several screens, different kinds of data flow between different UI components and views, validation, routing, etc. It must be something much more than the http://aurelia.io/docs.html app which shows static content
3. must use latest Aurelia or something close to the last version

Thanks

**Comment by wshayes** [comment]
Selventa Inc/OpenBEL.org -- BELMgr - http://belmgr.belframework.org  (demo site - not for production) - our production site is proprietary since it holds proprietary content.  Not a huge application or great example of Aurelia design principles since it is our first (actually second use - first use was a minor enhancement to a Django app) project.

**Comment by mikefills** [comment]
Thank you. 
It's a nice example but I'm really looking for web apps in production.

**Comment by wshayes** [comment]
Thanks for the reminder to update our Welcome msg.

**Comment by atsu85** [comment]
I used pre-alpha version of Aurelia (from June 2015) to create an internal application for local governments of the republic of Estonia (it is not available for public). Aurelia has evolved a lot since then, but to my mind mostly with plugins, nice-to-have features, tooling and architecturally (for example simplified APIs for really-really advanced use cases). There were only 2 small issues (with that version) i had to work around and they didn't compromise my day-to-day productivity at all. Comparing the framework with Angular, it was even back then a breath of fresh air, and it would be hard to go back to that again.
For now it is certainly my number 1 pick for SPA framework. Aurelia team has done a great work with both code and community!

**Comment by mPisano** [comment]
My first exposure to a SPA framework was with Durandal and BreezeJs developing a logistics application for use by our international offices, agents and customers.  We launched version 1 and like many was sad when Rob left and cheered when he returned with Aurelia. We have been working with Aurelia from the early beta’s porting the Durandal version over to version 2. We are now 100% complete with the port; The code is so much cleaner now and have added additional features that were difficult to do with prior.  

I would love to show the newer version, off but it’s not public and requires a login, but I will say I am very happy with the structure, Syntax and stability of Aurelia.  For 35 years I have programmed in every language since Pascal and finally think Java Script is ready. (or I’m ready for it).
We are now waiting for the RC bits while we finish bundling and prepare to launch. 

Many thanks to Rob and the team (especially @jdanyow) for this innovative and intuitive environment, I wish them and Aurelia great success.

**Comment by EisenbergEffect** [comment]
Awesome stuff all! Thanks for the encouragement!

**Comment by mikefills** [comment]
@mPisano Borland's Pascal, that's funny.

OK, so still no example of a web app product built with Aurelia which I can check-out. Maybe someone will see this post eventually and chime in.

I'm looking for:
1. a real product/service, so not a demo or POC
2. some complexity in the data flow, must have several screens, different kinds of data flow between different UI components and views, validation, routing, etc. It must be something much more than the http://aurelia.io/docs.html app which shows static content
3. must use latest Aurelia

It's nice to see support and appreciation for Aurelia, but the reality is it all comes down to see and experience real products and apps built with Aurelia.

I think having a list of apps done with Aurelia posted right on https://github.com/aurelia/framework is in Aurelia's best interest. Just a thought..

**Comment by Thanood** [comment]
I've found http://elee.menu/ on reddit. :smile:

**Comment by stsje** [comment]
https://gist.run/ is a good example. I believe @jdanyow is the creator.

Sadly our app is for enterprise, so you won't be able to see more than the login page.

**Comment by EisenbergEffect** [comment]
Most of the apps that are in production today are behind closed doors or paywalls. For example, http://www.hogia.com/ has put 6 Aurelia apps in production, but you won't be able to see them because you probably aren't purchasing software related to public transportation. There are lots of examples of this right now as there are plenty of Aurelia apps being written for health care, insurance, the financial industry, etc. They are significant applications but the general consumer can't gain access to them. Of course, there are lots of internal applications being built as well, which you will never see. What you are looking for is an open consumer-facing product built with Aurelia. There are some in development, but not yet launched. Many developers of those sorts of apps were waiting for our official beta and only started their development in November when we launched that, so they aren't ready to release just yet.

We plan on doing some case studies and adding a section to the site on this in the future. Doing the case studies are tricky because internal corporate politics are involved. In several cases, with large companies, I can't even tell you the names of the companies that are using Aurelia.

**Comment by jdanyow** [comment]
@mikefills - Public app with a decent amount of complexity [here](https://gist.run) (service workers, embedding, integration with APIs, etc)- it's not a LOB app so probably not quite what you're looking for.  Another public app that does LOB sorts of things that you're looking for is [here](http://jdanyow.github.io/aurelia-breeze-northwind/) but it needs updating, etc.

I'm in healthcare, we have three apps in development [where I work](http://www.mediware.com/), all behind login screens, can't share the code.

>It must be hard to get popularity against other frameworks when these frameworks were created by big companies like Google and Facebook, main reason might be many people think that if it's coming from Google/Facebook it must be the best way to do it.

[Here's a good perspective on that](http://blog.nofluffjuststuff.com/2016/02/22/podcast-checking-out-aurelia-with-peter-pavlovich/).

**Comment by mikefills** [comment]
@EisenbergEffect why close this? Can you please leave it open so more people see this and might contribute?

**Comment by EisenbergEffect** [comment]
Sure

**Comment by mikefills** [comment]
@EisenbergEffect thank you. the 1st link does nothing for me. the 2nd link is nice, I will review closely. 
About the podcast interview: I don't want to go off-topic, but it didn't bring anything new. The guy clearly says "I’m just a guy playing around with this stuff. I’m not using Aurelia on any production applications.". The whole interview is repeating exact same story and ideas from Rob's interview on Channel 9 at DEVintersection and his presentation at NDC London. 

@jdanyow thank you, I watche d

**Comment by EisenbergEffect** [comment]
@mikefills Here's something someone posted in the gitter room today: https://www.youtube.com/watch?v=yLs4_X7NX-U

**Comment by EisenbergEffect** [comment]
They said they built it in just a few days.

**Comment by EisenbergEffect** [comment]
There's lots of stuff going on, including from a number of companies you would know, I just can't show it to you unfortunately.

**Comment by fractal76** [comment]
About 5 days, including the developer having to figure out some business logic and twiddling with layout etc...

**Comment by mikefills** [comment]
@EisenbergEffect looks nice.
Yeah you mentioned about that before. But truth is, popularity is very important when choosing a framework. There are many people in the world right now looking to choose a framework for next project. Like me, some of them need to see something already done.

@fractal76 what you mean, you're the developer of  https://www.youtube.com/watch?v=yLs4_X7NX-U ?

**Comment by devmai** [comment]
This could be a good start to get in touch what Aurelia can do. It should be extended. 
[All-in-one demonstration app of gbegerow](https://github.com/gbegerow/aurelia-gbegerow-samples)
From that one I can imagine a lot stuff for a example showcase.

**Comment by iSkore** [comment]
@mikefills - same here. One of the biggest things for me is seeing examples of what the framework *can do* & *has done*. It's (probably) safe to say the Durandal and Aurelia websites are made using Aurelia, but with them looking nearly the same, for good branding purpose, this doesn't give visitors a good understanding of what it can produce. I saw **many** articles saying "screw Angular, go Aurelia!" But after half an hour on the Aurelia site, I didn't see a final "case study," or "other companies did this". Only examples I have seen were on this thread, so thank you. While I still indend on going with Aurelia, it would've been much better when telling my tech team to switch to this, too. They all brought up "what has it done" after 5 minutes of investigation.

I believe this mysterious *Project X* will be a marketplace addon similar to RapidWeaver's marketplace. That would be a very good move for a framework relatively shrouded in this Angulr VS Ember VS Aurelia VS Whatever turmoil we are all in right now. A move like this would justify the "top secret-ness" we are all awaiting.

Hope to see a bit more in the future please!

**Comment by EisenbergEffect** [comment]
FYI We are planning case studies. It takes time because we have to get permission from companies (legal, marketing, etc.) do the case study and then put together the content for our own site and blogs. Then it has to be approved again by the company (legal and marketing).

**Comment by iSkore** [comment]
@EisenbergEffect, thank you for the quick response. I understand it takes time, I know there are a lot of things going on in the framework religm right now! Any comments on Project X? 😬

**Comment by EisenbergEffect** [comment]
Project X is a generic name for a collection of things we are working on. We've announced Aurelia Interface...but that is not all we have planned for this year :) I can neither confirm nor deny any suspicions :laughing:

**Comment by iSkore** [comment]
Hahaha alright then! I suppose keeping closed alpha (and beta? 😢) is probably better. Seems like every major player is in some sort of transition.

**Comment by iSkore** [comment]
Anyway, thank you for your time! I am looking forward to seeing more from this framework and the addons. My team would certainly benefit from having a stable, forward thinking framework like this. One that is a major player with good support we can get really comfortable with that doesn't make a move like Angulr 1.x to 2. That's quite absurd to us. Have to change way too many things. This looks like the one to me. We looked into ember and comet and still aren't particularly happy. So THANK YOU!!!

**Comment by stefan2410** [comment]
We have in production the Admin portal, built in Aurelia for a product  running in call centers (desktop clients) in the Parking Industry.  We started October middle  and it works in production from end January.

**Comment by sensedeep** [comment]
We're developing a substantial management portal for a cloud-based intrusion detection service for a new startup SenseDeep. We've used Angular heavily in the past, but the elegant simplicity of Aurelia is compelling. So far so good in our development with no major issues we could not solve. We plan to launch 2nd half of this year.

**Comment by niieani** [comment]
I have developed a Slack/Gitter-clone using Aurelia and RethinkDB that uses GPG encryption for all conversations and file uploads; currently in production for internal use. That's quite a bit of complexity. I'll make it open source once I'll clean up the code.

**Comment by glen-84** [comment]
I just asked a similar question on Gitter: `Are there any non-trivial open-source Aurelia apps?`

It would really be great if there was an official "reference implementation" of a LOB or "Admin Control Panel"-type application, with multiple sections showing forms (with API back end), validation, hierarchical routing, etc. as well as some cross-cutting items such as authentication and internationalization.

**Comment by opcodewriter** [comment]
@glen-84 yep, exactly what I had in mind too.
An open source kitchen-sink app but covering several common scenarios like you mentioned (multiple screens, hierarchical, routing, form validation, authentication, etc)

An open source repo doubled by a series of blog posts would help I think to skyrocket the interest.

**Comment by alsoicode** [comment]
+1 For an example of hierarchical routing

**Comment by EisenbergEffect** [comment]
So, there's an app I've been wanting to build (or rather re-imagine) for a very long time. It's not an LOB. It's related to tabletop gaming. It's an example of a very complex application. I've been thinking about building the application as open source.

It's a significant effort though. I don't think I could do it without financial compensation. So, here's what I'm wondering. Would people be interested in backing a Kickstarter? The Kickstarter wouldn't actually be primarily directed at Aurelia devs but at people who would want to use the gaming product. However,  if it were open source, it could provide a tremendous learning resource for the Aurelia community. Also, I could create Kickstarter backer levels that included regular video code reviews and things like that for those who want a little more educational content.

There are lots of possibilities. I'm just throwing this out there because I'd love to build this particular app and I'm wondering if I could kill two birds with one stone...and wondering if the community would be interested in helping to fund something like this to be done in the open.

Frankly, if this sounded appealing, our core team could build a number of products, funded by Kickstarter, that we could later sell to average consumers, but make open source as learning resources. Open source is extremely expensive, so I'm always looking for new and interesting ways to sustain the project financially while enabling the team to do more and more for the community.

I'm just brainstorming. Let me know if you have any thoughts on this. Finding ways to produce more real-world example apps is important to us. I just thought...what if it wasn't an example at all. What if it was just an real, non-trivial product, created as open source?

**Comment by alsoicode** [comment]
+1 from me

The more examples I can put in front of the people who I have to convince
to let me use Aurelia, the better. I'm willing to chip in financially to
help make that easier.

Everyone I show Aurelia to always says the same thing: "Wow. It's so
simple." Then I show them what it can do, and their eyes light up,
especially if they're coming from an Angular background and they see how
much less code they have to write.

I'll buy a T-Shirt, stickers, mouse pad, books - anything to help.

**Comment by wshayes** [comment]
Fantastic idea about the game/kickstarter/opensource Aurelia example.  Also, we'd love more opportunities/approaches to send money your way.  I'm eagerly awaiting the Aurelia Interface partially to be able to financially support Aurelia.

**Comment by glen-84** [comment]
> Frankly, if this sounded appealing, our core team could build a number of products, funded by Kickstarter, that we could later sell to average consumers, but make open source as learning resources.

I'd be interested in something like this (and most-likely willing to make a small personal contribution/donation), but more so for a LOB-style application, as these are more likely to cover the topics/aspects that will assist me in writing my own applications.

**Comment by EisenbergEffect** [comment]
@glen-84 I'd love to hear what you would most like to see, in terms of features for an LOB app.

**Comment by glen-84** [comment]
@EisenbergEffect,

@opcodewriter and I have already mentioned most of the important parts. To summarize, we're looking for a real-world application that demonstrates everything that Aurelia has to offer. We're looking for guidance and best practices in implementing more complex aspects of an application.

The `skeleton-navigation` application is very much contrived, and is not developed in a way that can easily scale (putting all views and view models in a single folder, for example). It shows just the basics of what is possible.

A reference/best practices application might showcase:

* Use of TypeScript, the typings installer, how to deal with missing typings, etc.
* An improved folder structure where sections are separated into different folders.
    * (see an example of such a structure [here](https://github.com/aurelia/framework/issues/398))
* How to handle hierarchical navigation, generating breadcrumbs, etc.
    * It is unclear to me how you are supposed to make use of child routers, and whether something like [this](https://github.com/jdanyow/aurelia-breeze-northwind/blob/master/src/orders/orders-section.html) (now outdated), is the recommended pattern.
    * This is something that I was a bit stuck on recently, and it seems that some work still needs to be done (f.e. https://github.com/aurelia/router/issues/90).
    * I came up with a temporary solution using the `settings` key of a route to reference a parent route, and generating breadcrumbs using this information.
* Per-section index pages with tables of items with bound select/deselect all checkboxes, and possibly sorting, filtering, and paging.
    * This could show API interaction with the Fetch client, as well as local `repeat.for` value-converters/piping, etc.
* Bound forms with validation and possibly multi-step forms.
    * Different types of bindings for various form element types.
    * Reverting changes, cancelling nav, etc.
    * I understand that the validation library is still in development.
* Internationalization of views including translation and date/number formatting, etc.
    * Might be an area to show a real-world use of binding behaviours, etc.
    * Side note: I might look at something like [messageformat.js](https://github.com/SlexAxton/messageformat.js/) or [formatjs](http://formatjs.io/) instead of i18next, as I *think* that they are more aligned with browser standards.
* Perhaps a dashboard showing some custom elements and/or attributes.
* Authentication/user system showing integration with the router and use of setRoot to display a login page, etc.
* Use of the event aggregator in some meaningful way.
* Other Aurelia functionality like resources/features/plug-ins, etc.

You might argue that most of this is covered in documentation or sample applications, but the key points are:

* Real-world, not contrived.
* Always kept up-to-date with the latest releases.
* Best practices.
* Can be used as the starting point for a new application.
* Shows all available functionality.
* Open source.
* Possibly built up as a tutorial in the form of a series of blog posts (that are also kept up-to-date).

Some of the benefits:

* Dog-fooding: Developing and maintaining such an application will make it easier to identify issues and limitations in the framework.
* Clarity: New developers will immediately know what is and isn't possible, and the recommended way of implementing specific functionality.
* Documentation: The application will serve as a form of documentation, and should reduce the number of "how do I do this?"-type questions.
* Increased interest: With detailed blog posts covering the development of the application step-by-step (including the "whys" – why do we do it this way? performance/flexibility/etc.), there will likely be increased interest in the project, as many questions will be answered by example, leading to a better understanding of the framework.

Very few frameworks have done something like this, and I always miss it. I don't like having to "guess" best practices or intended usage, or pick from 5 or 10 different ways of doing something. Show me how you, as a contributor, would use your framework.

It could be a WordPress clone, or a StackOverflow clone, or a Discourse clone – the important part is having a larger application broken down into different sections/areas. These sections could even be installed as separate npm/jspm packages, but they should at least be separated in terms of folder structure, so that additional sections can be easily added.

I hope this helps, and keep up the great work with Aurelia. :+1:

**Comment by johnmanko** [comment]
@glen-84 Sounds like a great opportunity for you to provide something which benefits the community. The core team, afterall, is busy building a framework. ;)

**Comment by glen-84** [comment]
@johnmanko The whole point is for this to be done by someone with a deep understanding of the framework. I also have a full-time job and not much free time. :disappointed:

**Comment by gamefundas** [comment]
For my team an Admin Template (sample website with all web components and examples throw in the mix) always has been a good kickstart for projects. They contain best practices and examples that helps developers understand the framework usage and start building the application much faster. Rubix helped us greatly when we piloted our ReactJs based project. Currently searching for Aurelia based Admin templates yields no results, purposely ignoring the lone on out there which has been inactive for over a year now.

**Comment by fragsalat** [comment]
I'm not sure if I can say the name but I'm on introducing Aurelia into my company with almost 10000 employees and around 1000 in tech. Because I'm preparing the paper for tech review I found this issue.

Also I'm on migrating an app, which our team overtook, from Angular 1.4 to Aurelia. I'm in love with Aurelia since the first Alpha phase and really want to push it forward with that.

PS: It would be cool if you could review my document to introduce Aurelia and give suggestions how to improve: https://gist.github.com/fragsalat/adf50a1f1750cba04e71cd89b3d74ccc

**Comment by thelgevold** [comment]
I don't have an impressive Aurelia app to show, but I built a commenting component using Aurelia for my blog here: http://www.syntaxsuccess.com/viewarticle/comment-component-in-aurelia

The point of this is not to showcase a fancy UI (which this is not), but I think some of the bundling results might be interesting. 

I tried to tweak the bundle sizes until I got to what I consider production grade sizes... 
The main bundle (aurelia.js) ended up at 85k after gzip. I think that is pretty good. 

Granted I am only using data binding, templates and the fetch client, but still...

**Comment by doktordirk** [comment]
@fragsalat might add the awsome Aurelia Link Collection under the link sections for all those Blogs and resources

**Comment by riddla** [comment]
Chiming in here with some rough ideas and thoughts: I am frequently searching for code samples and mini application showing best practices regarding Aurelia. I know that @jdanyow and others are very helpful and often provide those kind of examples via gist.run. But is there a kind of repository/collection of these links? As we all know, though the Gitter archives are indexed, the information there is hard to find in hindsight.
I was always a fan of http://fdietz.github.io/recipes-with-angular-js/ when I was working with Angular 1, a curated list of (runnable) examples and best practices.
I also had the idea, that all the code snippets/samples that are shown in the official docs cheatsheet could be made available in a runnable format respectively via gist.run.

**Comment by AshleyGrant** [comment]
@riddla if you check out the docs, you'll see that a lot of the samples are runnable: http://aurelia.io/docs.html#/aurelia/binding/1.0.0-beta.1.3.5/doc/article/binding-binding-behaviors/2

**Comment by riddla** [comment]
@AshleyGrant: Oh, I see. But I find it often more convenient and better to learn from code that is presented "in context". Can't explain myself better, I can just easier look behind the code curtain and therefore debug and learn from something like https://gist.run/?id=7542e061bc940cde506b.

**Comment by AshleyGrant** [comment]
We have to strike a balance in the docs that gives runnable demos, but doesn't break the flow of the documentation itself. 

As to the overhaul, I believe Rob mentioned something regarding this to me at some point recently, but I cannot remember what it was he said :-) 

(I'm not trying to be cheeky, I really can't remember)

**Comment by EisenbergEffect** [comment]
Most of the work on the overhaul is not in new features, per se, but in transforming the docs architecture into a hub architecture for the community. Once this foundational work is done, it will be much easier for us to add the features that the community is asking for. So, think of it as an initial infrastructure upgrade that then allows us to do lots of cool stuff in the future, and to do that work more quickly as well.

**Comment by riddla** [comment]
@EisenbergEffect, thank you for the insights, it reads very promising. Do you think something like https://github.com/aurelia/documentation/issues/194 (Dash docset for Aurelia) will be easier in the future? Would love to help out there, when the right foundation is available.

**Comment by EisenbergEffect** [comment]
@riddla I don't know much about Dash docsets, but all our articles are available in markdown format on github today. All our apis are available as json documents on github today. If you can work with those formats, then you should be able to generate other documentation formats today.

**Comment by thelgevold** [comment]
My experience is that people seem to like component demos accompanied by a short blog post. I might take one or two of my existing blog demos and port them to Aurelia.

**Comment by ohjames** [comment]
kchomp.co is my company's aurelia web app. Should hit our first public milestone in the next two weeks.

**Comment by forestmarie** [comment]
@EisenbergEffect you can contact me for a case study. We are nearing code complete for v1 and we started a year ago. We obviously can't open source our code but our code base is pretty sizable now. 800 client side unit tests and about 50+ distinct views now with a good amount of custom attributes, components and value converters.  Hands down most developer friend JS framework we have used.

**Comment by stsje** [comment]
@forestmarie I would be very interested in knowing how you structure your application (directories) :).

**Comment by ohjames** [comment]
I'd be interested in hearing from anyone how long they have to wait while Aurelia is loading on mobile. Our project is 98kb bundled/minified and takes 3-4 seconds on a Samsung Galaxy S6.

**Comment by vdangwal** [comment]
Some Screen shot from our project. Its a LOB app with n-tier architecture. Aurelia + KendoUI + Materialize Css + Asp.Net Web API + SQL. 
![aurelia_screen1](https://cloud.githubusercontent.com/assets/6218372/16130191/08db5a7a-33d6-11e6-9d2b-f1e7a2f3ba9a.PNG)
![aurelia_screen2](https://cloud.githubusercontent.com/assets/6218372/16130192/08db990e-33d6-11e6-8f8b-c9775c42521e.PNG)
![aurelia_screen3](https://cloud.githubusercontent.com/assets/6218372/16130193/08dbbc90-33d6-11e6-8a11-394090fd71a6.PNG)

**Comment by vdangwal** [comment]
Its a first draft version of app and I am not a very good HTML developer. :)

**Comment by Thanood** [comment]
@vdangwal Looks very nice! :+1: 
Are you using vanilla KendoUI and materialize or one (or two) of the Aurelia bridges? :smile:

**Comment by vdangwal** [comment]
Bridges for both of them :) And if you remember, you helped me out on some issues with Materialize :+1:

**Comment by Thanood** [comment]
Yep, I remember.. The datepicker issue. :smile:

**Comment by vdangwal** [comment]
Yep.

**Comment by dpinart** [comment]
@ohjames We're alsp facing very, very long startup times on mobile devices. About 15 seconds in a Samsung Galaxy tab. Once app is loaded performance is not bad

**Comment by brandonseydel** [comment]
@SPARTAN563 Do you want to add the application you guys are building here?

**Comment by SPARTAN563** [comment]
Sure, http://farmtracksa.com - it's in production, uses the latest version of Aurelia (give or take a week depending on priorities and workload) and is built using TypeScript+JSPM. It was originally based on the Skeleton Navigation project however its architecture has since been significantly changed and includes a couple of neat features such as:

1. Built in authentication and permissions management through the use of routing pipeline injection, router settings, the DI container and filters (for conditional display: `if.bind="'permission/name' | can"`).
1. A customized version of the `aurelia-http-client` (using inheritance) which enables it to "mock" a strongly defined API by parsing an API Blueprint JSON output - removing the need for a server during testing.
1. A strong emphasis on DRY, making extensive use of components, globally registered resources etc.
1. A translation framework using i18next (prior to `aurelia-translation` being available, and adopting many of the same design principles, including some which I prefer for the time being).

Unfortunately it isn't open source and you will need to register to view the demo right now (we're working on putting together a static data version, but just haven't had the time yet) but please feel free to ask if you've got any further questions.

**Comment by stefan-sopic** [comment]
I'm currently working on a freelancing platform (http://fairlance.io), based on aurelia ES6 and gulp skeleton. Not really complex (so far) but multiple screens and a bit different code organization. https://github.com/fairlance/frontend

**Comment by hadrienl** [comment]
Hi,

You can try https://reverb.re

**Comment by emcga** [comment]
Aurelia-ui-framework is a UI framework that I'm currently using for an internal LOB application my company is developing. It's pretty nice to work with, and its developer is friendly and responsive to all questions/concerns. Also, Aurelia itself has been an absolute pleasure to work with. Thanks @EisenbergEffect and your team of devs for such an amazing framework! I convinced my team to switch to Aurelia over a year ago and have not been disappointed.

**Comment by utrolig** [comment]
If anyone is still wondering about stuff like this, I can probably make a couple examples of how I structure my applications in aurelia. I've made a couple of production-applications the last month, none of them huge, but I've found Aurelia to be much easier and intuitive to use than other frameworks.

I usually use the Aurelia-CLI for all my projects, and setup my folder-structure like this.

```
src/
    components/
        header/
            header.html
            header.ts
            header.scss
        nav/
            nav.html
            nav.ts
            nav.scss
    resources/
    views/
        dashboard/
            sub-page/
                sub-page.html
                sub-page.scss
                sub-page.ts
            dashboard.html
            dashboard.ts
            dashboard.scss
            dashboard.routes.ts
            dashboard.service.ts
    app.html
    app.ts
    app.constants.ts
    app.routes.ts
    enviroment.ts
    main.ts
```

### Components
In this folder I create all my "global" components, which don't belong to any specific view.

### Views
In this folder I create all the views, and sub-views.
I define all the routes separately by doing it like this:

 dashboard.routes.ts |
--- |

```typescript
export const DashboardRoutes = {
    route: 'dashboard',
    name: 'Dashboard',
    nav: true,
    data: {
        icon: 'dashboard',
    },
    moduleId: 'views/dashboard/dashboard'
}
```

I then import all the routes in app.ts with the following code.

```typescript
import { RouterConfiguration } from 'aurelia-router';

import { DashboardRoutes } from '.views/dashboard/dashboard.routes';

export class App {
    
  configureRouter(config: RouterConfiguration): void {
    config.title = 'Stian App';
    config.map([
      {
        route: '',
        redirect: 'home',
      }
    ]);
    config.mapRoute(DashboardRoutes);
  }
}
```

### Nav
When using the nav component, I have a nav.service as well, which broadcasts events about its state (open, closed) which the main application subscribes to, to be able to resize the main content-div based on which components are taking up screen-space. After resizing the main content-div I also have to redraw google maps. 

Let me know if any of this stuff is actually useful, and if there's anything else I can contribute with. I've only been using Aurelia for about a month, and programming for a year - but I've found this is a good way to separate code and keep it clean.

**Comment by damianof** [comment]
We worked for about 12 months (part-time) using Aurelia back in July 2015, went through some pain a couple of times with breaking changes when we updated to most recent versions. Then we froze our app on Aurelia beta a couple of months ago as we had to release our own beta (we finally did a few weeks ago). You are welcome to sign in using a Twitter account at [www.entensive.com](url) to check it out. Please mind, we are currently gathering feedback from beta users and are not planning to upgrade to Aurelia 1.0.0 yet, but that will come in a few months.

**Comment by hadrienl** [comment]
@damianof "Password must contain at least one upper-case, one lower-case, one number, and one symbol, and must be between 8 and 20 chars." =_= WHY 20 chars MAX??

**Comment by damianof** [comment]
@hadrienl The min and max length are actually driven from configuration. We can make it whatever we want, but early we made a decision that we should have a max. Do you feel 20 is too low?

**Comment by SPARTAN563** [comment]
@damianof might be worth having a gander through https://blogs.dropbox.com/tech/2012/04/zxcvbn-realistic-password-strength-estimation/ in which many of those points are touched on. We've integrated zxcvbn with our Aurelia app and had great results.

**Comment by hadrienl** [comment]
@damianof there should never be a maxlength for a password. A good password is not a password with special chars, but a very long one. Read the linked xkcd.

**Comment by ohjames** [comment]
> early we made a decision that we should have a max

You made a mistake then, the great thing about this one is it's easy to fix.

**Comment by damianof** [comment]
@ohjames thank you for pointing that out. @hadrienl  thank you for your feedback. We'll keep the max length though, maybe increasing it a bit. But we don't like the idea to have it open to infinite length.

**Comment by damianof** [comment]
@SPARTAN563 thank you for your feedback as well

**Comment by ohjames** [comment]
@damianof "Not liking the idea" isn't a reason to do something in computer science, scientific principles should be applied.

**Comment by damianof** [comment]
@ohjames Thank you again.Ultimately, each business is in charge of handling its own risk and decisions as its life depends on it. I definitely appreciate your feedback, but I hope you understand we are ultimately in charge of our decisions. I'm glad you feel strongly about following principles. That is a good thing.

**Comment by ohjames** [comment]
Feeling strongly about principles, no, autistic desire for order, yes. I'm now going to pretend that you don't exist so I can continue my day without feelings of horror ;)

**Comment by atsu85** [comment]
@utrolig, regarding Your sample project structure with scss files next to each component. Are You manually scoping the scss content so it would affect only that specific component or have You automated it using build pipeline to do smth like
1) extract style elements from Aurelia custom element templates
2) wrap the extracted styles with custom element tag name (to scope the styles to specific custom element)
3) feed the result to scss, to create single css file?

For example if i have a file:
src/my-demo-ce.html:
```html
<template>
    <div>style me!</div>
    <style>
        div {
            background-color: red;
        }
    </style>
</template>
```
then the build tool would copy templates from src and produce smth like:
build/my-demo-ce.html:
```html
<template>
    <div>style me!</div>
</template>
```
and
build/styles.scss:
```
my-demo-ce { /* WRAPPER: based on the custom element tag name to target elements from specific custom element*/
    div {
        background-color: red;
    }
}
/* ... syle definitions extracted from other custom elements */
```

*Note, I also [asked this question through Gitter channel](https://gitter.im/aurelia/Discuss?at=57d9ba5edf21a75730fa28da), but didn't get a positive answer, even tough is suspect someone has already implemented it (or smth conceptually similar to it) ;)*

**Comment by hadrienl** [comment]
@atsu85 I don't really understand the goal of your asking… It will only works once since my-demo-ce.html will have its style tag removed and you'll have to write your scss file. So, what you're asking for is just a way to automatically create the scss file when you create the html file, a task which is very simple to do manually…

**Comment by utrolig** [comment]
@atsu85 I haven't done anything custom in regards to scoping/encapsulation of the styles.. I just make sure to name my styles to avoid issues. 

So for example sub-page.scss would have all it styles called 

.sub-page-header
.sub-page-element
.sub-page-element-header
.sub-page-element-body

**Comment by ohjames** [comment]
Can you remove "kchomp.com" from "builtwithaurelia.com", we've had so many problems with Aurelia that we're porting the project to Angular 2.

**Comment by damianof** [comment]
@ohjames funny, as for the past year it has been the other way around for me. Angular, especially for rendering performance, has lots of issues.

**Comment by utrolig** [comment]
@ohjames could you elaborate on some specifics?

**Comment by EisenbergEffect** [comment]
Closing this issue since we've got the new site up.

**Comment by EisenbergEffect** [comment]
@utrolig There is apparently a bug in list rendering which crops up in edge cases which only one or two people are experiencing. He happens to be one. If it was properly reported earlier, it would have been fixed, but it wasn't until today. We are committed to resolve whatever critical issues are reported, provided that we can reproduce them. If they aren't reported or aren't reproducible then there's not much we can do. We fixed all the list rendering issues we knew about either during the alpha or the beta period.

It's important to realize that open source is about community. Every member of our team, when they are working on Aurelia, is choosing not to spend time with their wives, kids, etc. or enjoy their own hobbies. Some people even choose to consult less in order to spend even more time on Aurelia. That's a big sacrifice, especially for those who have been doing this for years. We work hard and we believe in what we're doing. We also invite everyone else to join with us in making Aurelia great. Contributing to Aurelia by helping with issue reporting and fixing has a positive effect on thousands of companies all over the world. If you experience a bug, the best response is to consider how you can help out, not to lash out, throw a fit or "take your toys and go home" (the comment above was intentionally put here in order to get the attention of a lot of people and leave a negative sentiment). That's not the type of behavior we think is acceptable in our community. Instead, consider how a very small contribution can positively affect the lives of thousands if not hundreds of thousands of people.

**Comment by vdangwal** [comment]
Well said Rob. 👍👍👍

Get Outlook for iOS<https://aka.ms/o0ukef>




On Fri, Nov 11, 2016 at 12:59 PM -0500, "Rob Eisenberg" <notifications@github.com<mailto:notifications@github.com>> wrote:


@utrolig<https://github.com/utrolig> There is apparently a bug in list rendering which crops up in edge cases which only one or two people are experiencing. He happens to be one. If it was properly reported earlier, it would have been fixed, but it wasn't until today. We are committed to resolve whatever critical issues are reported, provided that we can reproduce them. If they aren't reported or aren't reproducible then there's not much we can do. We fixed all the list rendering issues we knew about either during the alpha or the beta period.

It's important to realize that open source is about community. Every member of our team, when they are working on Aurelia, is choosing not to spend time with their wives, kids, etc. or enjoy their own hobbies. Some people even choose to consult less in order to spend even more time on Aurelia. That's a big sacrifice, especially for those who have been doing this for years. We work hard and we believe in what we're doing. We also invite everyone else to join with us in making Aurelia great. Contributing to Aurelia by helping with issue reporting and fixing has a positive effect on thousands of companies all over the world. If you experience a bug, the best response is to consider how you can help out, not to lash out, throw a fit or "take your toys and go home" (the comment above was intentionally put here in order to get the attention of a lot of people and leave a negative sentiment). That's not the type of behavior we think is acceptable in our community. Instead, consider how a very small contribution can positively affect the lives of thousands if not hundreds of thousands of people.

**Comment by opcodewriter** [comment]
I'm the OP of this, it's pretty sad to see how it got completely off-topic, arguing on password design or SCSS or complaining about Aurelia. Why's so hard to have some common sense!?
I think having concrete examples of fully fledged Aurelia apps is helpful.

> Closing this issue since we've got the new site up.
@EisenbergEffect, I might need to catch up, what is the new site you mentioned, does it have a list of apps developed with Aurelia?

**Comment by Gheoan** [comment]
@EisenbergEffect refers to http://builtwithaurelia.com/.

**Comment by opcodewriter** [comment]
@Gheoan thanks, that's nice. That's even more appropriate than this thread on GitHub, and it's something I've been thinking about too.

**Comment by sensedeep** [comment]
FYI:

We’ve been working on our Aurelia app, which is a cloud based Intrusion Detection Service for most of the year and have had a very positive experience.

We are launching late December and having used other frameworks before, we are very happy with the result and the major part Aurelia has played.

Aurelia continues to save us lots of dev time.

thank you all


Michael O'Brien
SenseDeep Inc.
mob@sensedeep.com

**Comment by geea789** [comment]
Hi all. We have a website in production, without a lot of traffic which is good for now. 
[https://hopayacht.com](url)
it uses Aurelia framework and if you go by mobile you can see Aurelia with Framework7.
We have a lot of work ahead of us but if anyone has any questions feel free. 
Thanks for all the community around Aurelia which helps us to overcome different problems.

**Comment by opcodewriter** [comment]
@geea789 looks nice. 
How come your `dist/app-bundle-ca2c680a57.js` is 1.2MB ?

**Comment by geea789** [comment]
meaning its too small or too big?

**Comment by opcodewriter** [comment]
@geea789 not sure exactly, because I don't know all the functionality of the app, but since it's just JS, it looks pretty big to me.

**Comment by opcodewriter** [comment]
the whole app is 3.2MB which is too big if you ask me

**Comment by geea789** [comment]
@opcodewriter actually right now it has a lot of functionality. too much.
looking back i would separate it into multiple apps, each for different sections of the site. something like suggested here.
[http://stackoverflow.com/questions/33473457/multiple-aurelia-apps-on-one-page](http://stackoverflow.com/questions/33473457/multiple-aurelia-apps-on-one-page)

**Comment by Thanood** [comment]
It's not only minified JS, it's also html and css in `app-bundle` (and a lof of `System.registerDynamic` calls containing strings).
Btw.: I'm impressed you get it to load so fast with SystemJS.

**Comment by jdemeuse1204** [comment]
I know you are looking for a production app, but I have a test production app I can share.  I am doing iterative development in my spare time, so anyone can check back for updates.  

Link:  http://www.skygroundlabs.net/

Like I said, the app isn't 100% ready, I am still working on it, but here is the code for anyone to check out.  Fork it, share it, I don't mind.  There isn't any sensitive information in it.  I used the Aurelia ESNext Webpack Skeleton as a base.

Code:  https://github.com/jdemeuse1204/CreditMattersWebSite

Since this is a test site, I do not mind sharing login credentials for a test user.  They are:

U: james.demeuse@gmail.com
P: abAB12!@

**Comment by jdemeuse1204** [comment]
Looks like a few people tried to login and failed because of the two-factor authentication.  I disabled 2FA, feel free to try again.

**Comment by jpravetz** [comment]
http://console.playable.video is a public Aurelia application.

**Comment by pndewit** [comment]
I am working for [Drillster](https://www.drillster.com/) which is a smart learning system to increase and maintain knowledge. Drillster is an adaptive, assessment based learning tool.

I am currently migrating a huge legacy AngularJS application to Aurelia part by part. Some parts are already migrated and running Aurelia in production. Some of these are publicly available (e.g. https://www.drillster.com/store) and for other parts you need to log in. Creating an account is free though. We also have a mechanism to let integrators use parts of our application (widgets) in their own web applications, most of these widgets are built using Aurelia and documentation can be found here: https://www.drillster.com/info/developers/widgets.

Feel free to look around, create an account and practice or even create some content. Everything in the [Drillster Store](https://www.drillster.com/store) is publicly available. If you start practicing an item from there a Player widget is opened, this widget is also created using Aurelia. For example, you could start practicing this example Drill about cars: https://www.drillster.com/store/playable/AHYrVtULyqnN1S7gRWq45Q

We also have a mobile app available in the [Google Play Store](https://play.google.com/store/apps/details?id=com.drillster) and the [Apple App Store](https://itunes.apple.com/nl/app/drillster/id429175115?l=en&mt=8). Both use the same code base as the Player widget linked above, with a small Aurelia/Cordova based wrapper for displaying some extra items. For the apps you will need to create an account though.
